src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java

```diff
@@ -0,0 +1,223 @@
+            return;
+        }
+
+        EntityRef etranceSpawner = structureTemplateProvider.
+                getRandomTemplateOfType("GooeysQuests:fortressVillageEntrance");
+
+        BlockRegionTransform foundSpawnTransformation = findGoodSpawnTransformation(surfaceGroundBlockPosition,
+                etranceSpawner);
+        if (foundSpawnTransformation == null) {
+            return;
+        }
+
+
+        questToFoundSpawnPossibilityMap.put(quest, new FoundSpawnPossiblity(etranceSpawner, foundSpawnTransformation));
+        quest.send(new QuestReadyEvent());
+    }
+
+    private static class FoundSpawnPossiblity {
+        private EntityRef entranceSpawner;
+        private BlockRegionTransform transformation;
+
+        public FoundSpawnPossiblity(EntityRef entranceSpawner, BlockRegionTransform transformation) {
+            this.entranceSpawner = entranceSpawner;
+            this.transformation = transformation;
+        }
+
+        public EntityRef getEntranceSpawner() {
+            return entranceSpawner;
+        }
+
+        public BlockRegionTransform getTransformation() {
+            return transformation;
+        }
+    }
+
+    private BlockRegionTransform findGoodSpawnTransformation(Vector3i spawnPosition, EntityRef entranceSpawner) {
+        for (Side side: Side.horizontalSides()) {
+            BlockRegionTransform transformList = createTransformation(spawnPosition, side);
+
+            CheckSpawnConditionEvent checkConditionEvent = new CheckSpawnConditionEvent(transformList);
+            entranceSpawner.send(checkConditionEvent);
+            if (!checkConditionEvent.isPreventSpawn()) {
+                return transformList;
+            }
+        }
+
+        return null;
+    }
+
+    private BlockRegionTransform createTransformation(Vector3i spawnPosition, Side side) {
+        return BlockRegionTransform.createRotationThenMovement(Side.FRONT, side, spawnPosition);
+    }
+
+    @ReceiveEvent(components = FortressVillageQuestComponent.class)
+    public void onQuestStart(QuestStartRequest event, EntityRef quest) {
+        FoundSpawnPossiblity spawnPossiblity = questToFoundSpawnPossibilityMap.get(quest);
+        BlockRegionTransform spawnTransformation = spawnPossiblity.getTransformation();
+        if (spawnTransformation == null) {
+            return; // TODO report failure to client and gooey system
+        }
+
+        EntityRef entranceSpawner = spawnPossiblity.getEntranceSpawner();
+        entranceSpawner.send(new SpawnStructureEvent(spawnTransformation));
+    }
+
+    /**
+     * Free the memory once the quest is no longer loaded
+     */
+    @ReceiveEvent
+    public void onDeactivateQuestEntity(BeforeDeactivateComponent event, EntityRef questEntity,
+                                        PersonalQuestsComponent questsComponent) {
+        questToFoundSpawnPossibilityMap.remove(questEntity);
+    }
+
+    private Vector3i findSurfaceGroundBlockPosition(Vector3i position) {
```

Comment:
Being that this method (and maybe others) appears duplicated between quests maybe there's a way to centralize some of the code?